### PR TITLE
Sidekiq will now retry and add backtraces

### PR DESCRIPTION
**Pull request by williampratt**
Sidekiq will now retry and add backtraces even if the job is enqueued via another adapter.

Note: it does honor the retry and backtrace settings if already set

### CODE REVIEW

File: lib/sidekiq_bus/middleware/retry.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module SidekiqBus
+  module Middleware
+    module Server
+      
+      # ensure sidekiq will retry jobs even when they are enqueued via other adapters
+      class Retry
+        def call(worker, msg, queue)
+          msg['retry'] = true unless msg['retry']
```
Comment (bleonard) [inline lib/sidekiq_bus/middleware/retry.rb]: shouldn't we do this only for our worker class?

File: lib/sidekiq_bus/middleware/retry.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      end
+
+    end
+
+    module Client
+     
+      # ensure sidekiq will retry jobs even when they are enqueued via other adapters
+      class Retry
+       def call(worker_class, job, queue, redis_pool)
+         job['retry'] = Customer.current_id = true unless job['retry']
```
Comment (bleonard) [inline lib/sidekiq_bus/middleware/retry.rb]: What's this `Customer` class for?

File: lib/sidekiq_bus/middleware/retry.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+module SidekiqBus
+  module Middleware
+    module Server
+      
+      # ensure sidekiq will retry jobs even when they are enqueued via other adapters
+      class Retry
+        def call(worker, msg, queue)
+          msg['retry'] = true unless msg['retry']
+          msg['backtrace'] = true unless msg['backtrace']
```
Comment (bleonard) [inline lib/sidekiq_bus/middleware/retry.rb]: This is just always setting it to `true` - did you mean do `unless msg.has_key?('backtrace')` and similiar on others?

File: lib/sidekiq_bus/version.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 module SidekiqBus
-  VERSION = "0.5.7"
+  VERSION = "0.5.8"
```
Comment (bleonard) [inline lib/sidekiq_bus/version.rb]: maybe do an `0.5.8.rc` next.

File: lib/sidekiq_bus/middleware/retry.rb (hunk 2 above)
Comment (williampratt) [reply lib/sidekiq_bus/middleware/retry.rb]: lol, I'll fix that. Must have been a copy / paste issue.

File: lib/sidekiq_bus/middleware/retry.rb (hunk 1 above)
Comment (williampratt) [reply lib/sidekiq_bus/middleware/retry.rb]: No, I can explain tomorrow. It needs to be done in both.

File: lib/sidekiq_bus/version.rb (hunk 4 above)
Comment (williampratt) [reply lib/sidekiq_bus/version.rb]: ok

File: lib/sidekiq_bus/middleware/retry.rb (hunk 3 above)
Comment (williampratt) [reply lib/sidekiq_bus/middleware/retry.rb]: yep, will fix

File: lib/sidekiq_bus/middleware/retry.rb (hunk 1 above)
Comment (williampratt) [reply lib/sidekiq_bus/middleware/retry.rb]: Don't merge this yet. I'm going to try something now that I made a change in the initializer in metrics.

**Comment by bleonard** [comment]
looks good. thanks!

File: lib/sidekiq_bus/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module SidekiqBus
-  VERSION = "0.5.7"
+  VERSION = "0.5.8rc"
```
Comment (bleonard) [inline lib/sidekiq_bus/version.rb]: usually `.rc`

**Comment by gfoster** [comment]
Hey @bleonard what is outstanding on this that is preventing a merge?

**Pull request merged**